pkg/gui/controllers/local_commits_controller.go

```diff
@@ -924,6 +928,50 @@ func (self *LocalCommitsController) createFixupCommit(commit *models.Commit) err
 	})
 }
 
+func (self *LocalCommitsController) moveFixupCommitToCurrentBranchOfStack(targetCommit *models.Commit) error {
+	if self.c.Git().Version.IsOlderThan(2, 38, 0) {
+		// Git 2.38.0 introduced the `rebase.updateRefs` config option. Don't
+		// move the commit down with older versions, as it would break the stack.
+		return nil
+	}
+
+	if self.c.Git().Status.WorkingTreeState() != enums.REBASE_MODE_NONE {
+		// Can't move commits while rebasing
+		return nil
+	}
+
+	if targetCommit.Status == models.StatusMerged {
+		// Target commit is already on main. It's a bit questionable that we
+		// allow creating a fixup commit for it in the first place, but we
+		// always did, so why restrict that now; however, it doesn't make sense
+		// to move the created fixup commit down in that case.
+		return nil
+	}
+
+	if !self.c.Git().Config.GetRebaseUpdateRefs() {
+		// If the user has disabled rebase.updateRefs, we don't move the fixup
+		// because this would break the stack of branches (presumably they like
+		// to manage it themselves manually, or something).
+		return nil
+	}
+
+	headOfCurrentBranchIdx := -1
```

Comment:
This variable name suggests that it's the index of the current branch's head, but when I think 'head of current branch' I think 'the commit with an index of 0'. It's moreso talking about the head of the branch that 'owns' the commit. Not sure if there's any established lingo for that, but I'd be happy to use the term 'owner' and rename this to 'headOfOwnerBranchIdx'. At any rate I'd add a comment explaining what we're doing: finding the lowest branch in the stack that contains the commit.

Another term we could use is 'original'.

Reply:
I agree that "current" is not a good name. I guess I was thinking about the "current selected commit" or something.

Owner is a little bit better (not very much, tbh :) Changed to that in 9553e304b35e.

Another option that I considered was "target branch".